sdks/python/apache_beam/examples/inference/pytorch_language_modeling.py

```diff
@@ -0,0 +1,213 @@
+    standard torch.Tensor output. Instead, they can return a dictionary of
+    different outputs. To work with current RunInference implementation which
+    returns a PredictionResult object for each example, we must override the
+    `forward()` function and convert the standard Hugging Face forward output
+    into the appropriate format of List[Dict[str, torch.Tensor]].
+
+    Before:
+    output = {
+      'logit': torch.FloatTensor of shape
+        (batch_size, sequence_length, config.vocab_size),
+      'hidden_states': tuple(torch.FloatTensor) of shape
+        (batch_size, sequence_length, hidden_size)
+    }
+    After:
+    output = [
+      {
+        'logit': torch.FloatTensor of shape
+          (sequence_length, config.vocab_size),
+        'hidden_states': tuple(torch.FloatTensor) of
+          shape (sequence_length, hidden_size)
+      },
+      {
+        'logit': torch.FloatTensor of shape
+          (sequence_length, config.vocab_size),
+        'hidden_states': tuple(torch.FloatTensor) of shape
+          (sequence_length, hidden_size)
+      },
+      ...
+    ]
+    where len(output) is batch_size
+    """
+    def forward(self, **kwargs):
+      output = super().forward(**kwargs)
+      return [dict(zip(output, v)) for v in zip(*output.values())]
+
+  # TODO: Remove once nested tensors https://github.com/pytorch/nestedtensor
+  # is officially released.
+  class PytorchNoBatchModelHandler(PytorchModelHandler):
+    """Wrapper to PytorchModelHandler to limit batch size to 1.
+
+    The tokenized strings generated from BertTokenizer may have different
+    lengths, which doesn't work with torch.stack() in current RunInference
+    implementation since stack() requires tensors to be the same size.
+
+    Restricting max_batch_size to 1 means there is only 1 example per `batch`
+    in the run_inference() call.
+    """
+    def batch_elements_kwargs(self):
+      return {'max_batch_size': 1}
+
+  model_handler = PytorchNoBatchModelHandler(
+      state_dict_path=known_args.model_state_dict_path,
+      model_class=HuggingFaceStripBatchingWrapper,
+      model_params=model_params)
+
+  with beam.Pipeline(options=pipeline_options) as p:
+    text = (p | 'ReadSentences' >> beam.io.ReadFromText(known_args.input))
+    text_and_masked_text_tuple = (
+        text
+        | 'AddMask' >> beam.Map(add_mask_to_last_word))
+    text_and_tokenized_text_tuple = (
+        text_and_masked_text_tuple
+        | 'TokenizeSentence' >> beam.Map(tokenize_sentence))
+    text_and_predictions = (
+        text_and_tokenized_text_tuple
+        | 'PyTorchRunInference' >> RunInference(
+            KeyedModelHandler(model_handler)).with_output_types(
+                Tuple[str, PredictionResult])
+        | 'ProcessOutput' >> beam.ParDo(PostProcessor()))
+    combined_text = (({
+        'masked_text': text_and_masked_text_tuple,
+        'predicted_text': text_and_predictions
+    })
+                     | 'Merge' >> beam.CoGroupByKey()
```

Comment:
is this something that can be folded into PostProcess? isn't the masked_text already a part of predicted result?